hedera-node/src/test/java/com/hedera/services/state/migration/MigrationRecordsManagerTest.java

```diff
@@ -580,6 +580,37 @@ void traceabilityMigrationDoesNotAddStateChangesWhenSmartContractDoesNotHaveAnyS
                 sidecarRecords.build());
     }
 
+    @Test
+    void traceabilityMigrationDoesNotCreateMigrationRecordsForContractWithoutBytecodeInState() {
+        given(consensusTimeTracker.unlimitedPreceding()).willReturn(true);
+        given(networkCtx.areMigrationRecordsStreamed()).willReturn(false);
+        given(networkCtx.consensusTimeOfLastHandledTxn()).willReturn(now);
+        MigrationRecordsManager.setExpiryJustEnabled(false);
+        given(txnAccessor.getFunction()).willReturn(HederaFunctionality.ConsensusCreateTopic);
```

Comment:
Should this not be for Contract related transaction?
```suggestion
        given(txnAccessor.getFunction()).willReturn(HederaFunctionality.ContractCreate);
```

Reply:
The traceability migration is triggered only by a non-contract related transaction that changes state, i.e. all except `ContractCall`, `ContractCreate`, `EthereumTransaction`, since they themselves will have sidecars generated from their execution. That is why the transaction in the test is set as `ConsensusCreateTopic`.

We have [separate tests](https://github.com/hashgraph/hedera-services/blob/7140ff130efad054ac35d0f130f5364ed052d645/hedera-node/src/test/java/com/hedera/services/state/migration/MigrationRecordsManagerTest.java#L234-L280)that verify that these 3 transactions do not trigger the migration.